### PR TITLE
feat(route): add 机核网专题文章

### PR DESCRIPTION
**Pull request by kudryavka1013**
## 路由地址示例 / Example for the Proposed Route(s)

```routes
gcores/collections/64
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [X] 新的路由 New Route
  - [X] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [X] 文档说明 Documentation
  - [X] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/gcores/collection.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                const content = $('.story.story-show').html();
+                return {
+                    title: item.title,
+                    description: cover + content,
+                    link: articleUrl,
+                    guid: articleUrl,
+                };
+            });
+        })
+    );
```
Comment (TonyRL) [inline lib/v2/gcores/collection.js:95]: Please use the API of `/gapi/v1/collections/${collection}` and `/gapi/v1/collections/${collection}/originals` instead. They include both the feed metadata and full articles.

**Review by TonyRL — COMMENTED** [review]
Missing respective entries of `maintainer.js` and `radar.js`

File: lib/v2/gcores/collection.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            const articleUrl = `https://www.gcores.com/${item.url}`;
+            return ctx.cache.tryGet(articleUrl, async () => {
+                const itemRes = await got({
+                    method: 'get',
+                    url: articleUrl,
+                });
+
+                const itemPage = itemRes.data;
+                const $ = cheerio.load(itemPage);
+                let articleData = await got(`https://www.gcores.com/gapi/v1/${item.url}?include=media`);
```
Comment (TonyRL) [inline lib/v2/gcores/collection.js:35]: The JSON response of `res` in line 12 already contains the full articles (blocks) and their media (entityMap) in `data.attributes.content`. 

You don't need to visit `https://www.gcores.com/${item.url}` and `https://www.gcores.com/gapi/v1/${item.url}?include=media` thereafter.

Comment (kudryavka1013) [reply lib/v2/gcores/collection.js:35]: That means I need to format all content blocks into html by myself? I will take a look at `data.attributes.content`. Thanks

**Comment by Zzzzz-Haha** [comment]
大佬，首先感谢您做了这个路由~~请问这个路由做好了目前已经能够使用了吗？

**Comment by kudryavka1013** [comment]
@TonyRL Please check again, I think this time the code should meet your requirements. Thank you!!

File: lib/v2/gcores/collection.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                    // split string and insert HTML tag
+                    const strArr = text.split('');
+                    let isMedia = false;
+                    if (entityRanges.length) {
+                        entityRanges.forEach((entityRange) => {
+                            const { key, offset, length } = entityRange;
+                            const startIdx = offset,
+                                endIdx = offset + length;
+                            const { data, type } = entityMap[key];
+                            switch (type) {
```
Comment (TonyRL) [inline lib/v2/gcores/collection.js:54]: Please a default case if none of the types matches.

File: lib/v2/gcores/collection.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                                    break;
+                            }
+                        });
+                    }
+                    if (inlineStyleRanges.length) {
+                        inlineStyleRanges.forEach((inlineStyleRange) => {
+                            const { style, offset, length } = inlineStyleRange;
+                            const startIdx = offset,
+                                endIdx = offset + length;
+                            switch (style) {
```
Comment (TonyRL) [inline lib/v2/gcores/collection.js:78]: Please a default case if none of the styles matches.

File: lib/v2/gcores/maintainer.js [hunk 5]
```diff
@@ -2,4 +2,5 @@ module.exports = {
     '/category/:category': ['MoguCloud', 'StevenRCE0'],
     '/radios/:category?': ['eternasuno'],
     '/tag/:tag/:category?': ['StevenRCE0'],
+    '/collections/:collection': ['kudryavka1013'],
```
Comment (TonyRL) [inline lib/v2/gcores/maintainer.js:5]:
```suggestion
    '/category/:category': ['MoguCloud', 'StevenRCE0'],
    '/collections/:collection': ['kudryavka1013'],
    '/radios/:category?': ['eternasuno'],
    '/tag/:tag/:category?': ['StevenRCE0'],
```
Please insert new entries in alphabetical order.

File: lib/v2/gcores/router.js [hunk 6]
```diff
@@ -2,4 +2,5 @@ module.exports = function (router) {
     router.get('/category/:category', require('./category'));
     router.get('/radios/:category?', require('./radio'));
     router.get('/tag/:tag/:category?', require('./tag'));
+    router.get('/collections/:collection', require('./collection'));
```
Comment (TonyRL) [inline lib/v2/gcores/router.js:5]:
```suggestion
    router.get('/category/:category', require('./category'));
    router.get('/collections/:collection', require('./collection'));
    router.get('/radios/:category?', require('./radio'));
    router.get('/tag/:tag/:category?', require('./tag'));
```
Please insert new entries in alphabetical order.

**Pull request merged**